ql/src/java/org/apache/hadoop/hive/ql/QueryState.java

```diff
@@ -59,6 +60,11 @@
 
   static public final String USERID_TAG = "userid";
 
+  /**
+   * map of tables involved in the query.
+   */
+  private final Map<String, Object> tableMap = new HashMap<>();
```

Comment:
Can we make this a generic map so we can store things other than tables here? We will need this container also for other info that - in the absence of this new query state feature - we previously had to force into the conf, such as write commit info (jobID, vertexID, taskNum), CTAS info (is the query ctas? what's the ctas target table name), etc.

Reply:
Changed the naming of the map.